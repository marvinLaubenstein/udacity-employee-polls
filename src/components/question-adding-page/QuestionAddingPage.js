import React, { useState } from 'react';
import { connect } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { handleAddNewQuestion } from '../../actions/questions';
import Navbar from '../navbar/Navbar';
import './questionaddingpage.css';

const QuestionAddingPage = ({ dispatch }) => {
  const [firstAnswer, setFirstAnswer] = useState([]);
  const [secondAnswer, setSecondAnswer] = useState([]);
  const [error, setError] = useState(false);
  const navigate = useNavigate();

  const handleChange = (e) => {
    e.target.id === 'question1'
      ? setFirstAnswer(e.target.value)
      : setSecondAnswer(e.target.value);
  };

  const handleClick = (e) => {
    e.preventDefault();
    if (firstAnswer || secondAnswer) {
      setError(true);
    } else {
      dispatch(handleAddNewQuestion(firstAnswer, secondAnswer));
      setError(false);
      navigate('/');
    }
  };

  return (
    <div>
      <Navbar></Navbar>
      <div className="question-adding-page-title-wrapper">
        <h1 className="question-adding-page-title">Would you rather</h1>

        {error ? (
          <h3
            className="question-adding-page-errormessage"
            data-testid="error-title"
          >
            Please enter two answers!
          </h3>
        ) : null}
      </div>
      <div className="question-adding-page-answer-one">
        <label>Answer 1:</label>
        <input
          className="question-adding-page-input"
          data-testid="answer-one-input"
          type="text"
          id="question1"
          onChange={handleChange}
          data-testid="answer-one-input"
        ></input>
      </div>
      <div className="question-adding-page-answer-two">
        <label>Answer 2:</label>
        <input
          className="question-adding-page-input"
          data-testid="answer-two-input"
          type="text"
          id="question2"
          onChange={handleChange}
          data-testid="answer-two-input"
        ></input>
      </div>

      <button
        className="question-adding-page-button"
        onClick={handleClick}
        data-testid="submit-button"
      >
        Add them !
      </button>
    </div>
  );
};

export default connect()(QuestionAddingPage);
