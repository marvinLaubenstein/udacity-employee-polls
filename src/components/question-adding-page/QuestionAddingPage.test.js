import React from 'react';
import { Provider } from 'react-redux';
import reducer from '../../reducers';
import middleware from '../../middleware';
import { BrowserRouter } from 'react-router-dom';
import { createStore } from 'redux';
import QuestionAddingPage from './QuestionAddingPage.js';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('QuestionAddingPage', () => {
  let component;
  const store = createStore(reducer, middleware);
  beforeEach(async () => {
    component = render(
      <Provider store={store}>
        <BrowserRouter>
          <QuestionAddingPage />
        </BrowserRouter>
      </Provider>
    );
  });
  it('should match snapshot after render', () => {
    expect(component).toMatchSnapshot();
  });
  it('should handle no input', () => {
    var submitButton = component.getByTestId('submit-button');
    fireEvent.click(submitButton);
    expect(component.getByTestId('error-title')).toBeInTheDocument();
  });
  it('should handle just one input', () => {
    var inputOne = component.getByTestId('answer-one-input');
    fireEvent.change(inputOne, { target: { value: 'First answer option' } });
    var submitButton = component.getByTestId('submit-button');
    fireEvent.click(submitButton);
    expect(component.getByTestId('error-title')).toBeInTheDocument();
  });
});
