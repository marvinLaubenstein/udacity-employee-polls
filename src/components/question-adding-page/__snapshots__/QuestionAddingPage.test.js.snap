// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionAddingPage should match snapshot after render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <ul
          class="navbar"
        >
          <li>
            <div
              class="navbar-auth-user-item"
            />
          </li>
          <li>
            <ul
              class="navbar-link-list"
            >
              <li
                class="navbar-link-item"
              >
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="navbar-link-item"
              >
                <a
                  href="/add"
                >
                  New Question
                </a>
              </li>
              <li
                class="navbar-link-item"
              >
                <a
                  href="/leaderboard"
                >
                  LeaderBoard
                </a>
              </li>
            </ul>
          </li>
          <li
            class="navbar-logout-item"
          >
            <button>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LogoutIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
            </button>
          </li>
        </ul>
        <div
          class="question-adding-page-title-wrapper"
        >
          <h1
            class="question-adding-page-title"
          >
            Would you rather
          </h1>
        </div>
        <div
          class="question-adding-page-answer-one"
        >
          <label>
            Answer 1:
          </label>
          <input
            class="question-adding-page-input"
            data-testid="answer-one-input"
            id="question1"
            type="text"
          />
        </div>
        <div
          class="question-adding-page-answer-two"
        >
          <label>
            Answer 2:
          </label>
          <input
            class="question-adding-page-input"
            data-testid="answer-two-input"
            id="question2"
            type="text"
          />
        </div>
        <button
          class="question-adding-page-button"
          data-testid="submit-button"
        >
          Add them !
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <ul
        class="navbar"
      >
        <li>
          <div
            class="navbar-auth-user-item"
          />
        </li>
        <li>
          <ul
            class="navbar-link-list"
          >
            <li
              class="navbar-link-item"
            >
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="navbar-link-item"
            >
              <a
                href="/add"
              >
                New Question
              </a>
            </li>
            <li
              class="navbar-link-item"
            >
              <a
                href="/leaderboard"
              >
                LeaderBoard
              </a>
            </li>
          </ul>
        </li>
        <li
          class="navbar-logout-item"
        >
          <button>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LogoutIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
              />
            </svg>
          </button>
        </li>
      </ul>
      <div
        class="question-adding-page-title-wrapper"
      >
        <h1
          class="question-adding-page-title"
        >
          Would you rather
        </h1>
      </div>
      <div
        class="question-adding-page-answer-one"
      >
        <label>
          Answer 1:
        </label>
        <input
          class="question-adding-page-input"
          data-testid="answer-one-input"
          id="question1"
          type="text"
        />
      </div>
      <div
        class="question-adding-page-answer-two"
      >
        <label>
          Answer 2:
        </label>
        <input
          class="question-adding-page-input"
          data-testid="answer-two-input"
          id="question2"
          type="text"
        />
      </div>
      <button
        class="question-adding-page-button"
        data-testid="submit-button"
      >
        Add them !
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
